tool/tsh/db.go

```diff
@@ -62,10 +107,112 @@ func onListDatabases(cf *CLIConf) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	showDatabases(cf.SiteName, databases, activeDatabases, cf.Verbose)
+
+	showDatabases(cf.SiteName, displayUsersForDb, databases, activeDatabases, cf.Verbose)
 	return nil
 }
 
+// getUsersForDb finds allowed and denied users given user's effective roleSet and database in consideration. The result is a user-readable string.
+func getUsersForDb(roleSet services.RoleSet, database types.Database) string {
```

Comment:
Instead of implementing this in the CLI, this needs to be a part of the API. This way UI will also be able to reuse it for example. I would either make this a part of the GetDatabases API, or introduce a separate API endpoint.

Reply:
This makes it more complex to implement, but I can see the benefits. I'll look into this after v9 is out.

Reply:
@Tener I've been thinking, it probably doesn't even need to be an actual API - but we should at least move this function somewhere it can be reused e.g. to lib/services package. This way web API will also be able to use it since it has RoleSet and Database.

Reply:
@r0mant sounds reasonable; I generally agree this should move towards the application's core - either as a callable service or as part of the core library. I'll see which option feels better once I get back to this.